transports/transport-rest/transport-rest-client/src/main/java/org/apache/servicecomb/transport/rest/client/RestTransportClient.java

```diff
@@ -62,7 +77,16 @@ private static HttpClientOptions createHttpClientOptions() {
   }
 
   public void send(Invocation invocation, AsyncResponse asyncResp) throws Exception {
-    HttpClientWithContext httpClientWithContext = clientMgr.findClientPool(invocation.isSync());
+    URIEndpointObject endpoint = (URIEndpointObject) invocation.getEndpoint().getAddress();
+    HttpClientWithContext httpClientWithContext;
+    if (endpoint.isHttp2Enabled()) {
+      httpClientWithContext = clientMgrHttp2.findClientPool(invocation.isSync());
+      if (!endpoint.isSslEnabled()) {
+        ((HttpClientImpl) httpClientWithContext.getHttpClient()).getOptions().setHttp2ClearTextUpgrade(false);
```

Comment:
multi invocation will share the same httpClient instance when use http2
so there is concurrence problem here, right?

Reply:
http2 + ssl         = h2
http2 + without ssl = h2c
for the http2 client instance to work with h2c the http2ClearTextUpgrade need to be set to false

![image](https://user-images.githubusercontent.com/13465886/39046247-264d48f8-44b3-11e8-99d3-dafcd9bed953.png)